src/surrogates/inc/SurrogateBase.h

```diff
@@ -0,0 +1,52 @@
+//-----------------------------------------------------------------------bl-
+//--------------------------------------------------------------------------
+//
+// QUESO - a library to support the Quantification of Uncertainty
+// for Estimation, Simulation and Optimization
+//
+// Copyright (C) 2008-2015 The PECOS Development Team
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the Version 2.1 GNU Lesser General
+// Public License as published by the Free Software Foundation.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc. 51 Franklin Street, Fifth Floor,
+// Boston, MA  02110-1301  USA
+//
+//-----------------------------------------------------------------------el-
+
+#ifndef UQ_SURROGATE_BASE_H
+#define UQ_SURROGATE_BASE_H
+
+namespace QUESO
+{
+  //! Base class for surrogates of models
+  /*! Defines basic interface for using surrogates of models. The idea
+      is that we have some surrogate of the parameter-to-data map so that
+      it can be used in the likelihood classes. Subclasses will define the
+      particular surrogate model. Other classes will be used to build up the
+      surrogate from the user's model. */
+  template<class V>
+  class SurrogateBase
+  {
+  public:
+
+    SurrogateBase(){};
+
+    virtual ~SurrogateBase(){};
+
+    //! Method to return value given the parameter vector
+    virtual double evaluate(const V & domainVector) const =0;
```

Comment:
This looks as though we're only supporting surrogates of models that map `\mathbb{R}^n \to \mathbb{R}`, is that correct?  I.e., we're not currently supporting `\mathbb{R}^n \to \mathbb{R}^m`

I'm not suggesting implementing the multi-dimensional output case in this PR (if at all), I'm merely asking for purposes pertaining to my understanding.